src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrDataValidator.java

```diff
@@ -1,23 +1,131 @@
+    private static final Logger log = LoggerFactory.getLogger(OcrDataValidator.class);
+
+    public OcrValidationResult validate(FormType formType, List<OcrDataField> ocrData) {
+        List<String> duplicateOcrKeys = OcrFormValidationHelper.findDuplicateOcrKeys(ocrData);
+
+        if (duplicateOcrKeys.isEmpty()) {
+            List<String> errors = validateMandatoryFields(formType, ocrData);
+            List<String> warnings = validateOptionalFields(formType, ocrData);
+
+            return new OcrValidationResult(
+                warnings, errors, getValidationStatus(!errors.isEmpty(), !warnings.isEmpty())
+            );
+        } else {
+            String duplicateKeys = String.join(",", duplicateOcrKeys);
+            log.info("Found duplicate keys in OCR data. {}", duplicateKeys);
+
+            String errorMessage = String.format("Invalid OCR data. Duplicate keys exist: %s", duplicateKeys);
+            return new OcrValidationResult(emptyList(), singletonList(errorMessage), ERRORS);
+        }
+    }
+
+    private List<String> validateMandatoryFields(FormType formType, List<OcrDataField> ocrData) {
+        List<String> mandatoryFields = getMandatoryFieldsForForm(formType);
+        List<String> missingFields = OcrFormValidationHelper.findBlankFields(mandatoryFields, ocrData);
+        List<String> errors = OcrFormValidationHelper.getErrorMessagesForMissingFields(missingFields);
+
+        if (formType.equals(CONTACT)) {
+            String email = OcrFormValidationHelper.findOcrFormFieldValue(EMAIL, ocrData);
+            String phone = OcrFormValidationHelper.findOcrFormFieldValue(CONTACT_NUMBER, ocrData);
+
+            if (!errors.contains(EMAIL) && !isValidEmailAddress(email)) {
+                errors.add("Invalid email address");
+            }
+            if (!errors.contains(CONTACT_NUMBER) && !isValidPhoneNumber(phone)) {
+                errors.add("Invalid phone number");
+            }
+        }
+
+        return errors;
     }
 
-    public static OcrValidationResult validate(FormType formType, List<OcrDataField> ocrDataFields) {
-        //TODO validate OCr data
-        return new OcrValidationResult(emptyList(), emptyList(), SUCCESS);
+    private List<String> validateOptionalFields(FormType formType, List<OcrDataField> ocrData) {
+        List<String> optionalFields = getOptionalFieldsForForm(formType);
+        List<String> ocrInputFields = getOcrFieldNames(ocrData);
+
+        List<String> missingFields = OcrFormValidationHelper.findMissingFields(optionalFields, ocrInputFields);
+        return OcrFormValidationHelper.getErrorMessagesForMissingFields(missingFields);
+    }
+
+    private List<String> getMandatoryFieldsForForm(FormType formType) {
+        if (formType.equals(CONTACT)) {
+            return asList(
+                OcrFieldNames.ADDRESS_LINE_1,
+                EMAIL,
+                OcrFieldNames.POST_CODE,
+                OcrFieldNames.COUNTRY,
+                CONTACT_NUMBER
```

Comment:
are `EMAIL` and `CONTACT_NUMBER` not ocr field names?

Reply:
they are.. added static imports for all ocr fields